src/openms/source/ANALYSIS/ID/BasicProteinInferenceAlgorithm.cpp

```diff
@@ -0,0 +1,362 @@
+    bool treat_modification_variants_separately(param_.getValue("treat_modification_variants_separately").toBool());
+    bool use_shared_peptides(param_.getValue("use_shared_peptides").toBool());
+    bool skip_count_annotation(param_.getValue("skip_count_annotation").toBool());
+
+    String aggMethodString(param_.getValue("score_aggregation_method").toString());
+    AggregationMethod aggregation_method = AggregationMethod::MAXIMUM;
+
+    if (aggMethodString == "maximum")
+    {
+      aggregation_method = AggregationMethod::MAXIMUM;
+    }
+    else if (aggMethodString == "product")
+    {
+      aggregation_method = AggregationMethod::PROD;
+    }
+    else if (aggMethodString == "sum")
+    {
+      aggregation_method = AggregationMethod::SUM;
+    }
+
+    std::unordered_map<std::string, std::map<Int, PeptideHit*>> best_pep{};
+
+    // iterate over runs
+    std::unordered_map<std::string, std::pair<ProteinHit*, Size>> acc_to_protein_hitP_and_count;
+
+    processRun_(
+        acc_to_protein_hitP_and_count,
+        best_pep,
+        prot_id,
+        pep_ids,
+        aggregation_method,
+        aggMethodString,
+        use_shared_peptides,
+        treat_charge_variants_separately,
+        treat_modification_variants_separately,
+        skip_count_annotation);
+
+    //TODO Filtering? I think this should be done separate afterwards with IDFilter
+    //for all protein hits for the id run, only accept proteins that have at least 'min_peptides_per_protein' peptides
+  }
+
+  void BasicProteinInferenceAlgorithm::run(std::vector<PeptideIdentification> &pep_ids,
+                                           std::vector<ProteinIdentification> &prot_ids) const
+  {
+    //Size min_peptides_per_protein = getIntOption_("min_peptides_per_protein");
+    bool treat_charge_variants_separately(param_.getValue("treat_charge_variants_separately").toBool());
+    bool treat_modification_variants_separately(param_.getValue("treat_modification_variants_separately").toBool());
+    bool use_shared_peptides(param_.getValue("use_shared_peptides").toBool());
+    bool skip_count_annotation(param_.getValue("skip_count_annotation").toBool());
+
+    String aggMethodString(param_.getValue("score_aggregation_method").toString());
+    AggregationMethod aggregation_method = AggregationMethod::MAXIMUM;
+
+    if (aggMethodString == "maximum")
+    {
+      aggregation_method = AggregationMethod::MAXIMUM;
+    }
+    else if (aggMethodString == "product")
+    {
+      aggregation_method = AggregationMethod::PROD;
+    }
+    else if (aggMethodString == "sum")
+    {
+      aggregation_method = AggregationMethod::SUM;
+    }
+
+    std::unordered_map<std::string, std::map<Int, PeptideHit*>> best_pep{};
+
+    // iterate over runs
+    std::unordered_map<std::string, std::pair<ProteinHit*, Size>> acc_to_protein_hitP_and_count;
+
+    for (auto &prot_run : prot_ids)
```

Comment:
why not replace the whole function with 

```suggestion
    for (auto &prot_run : prot_ids) run(pep_ids, prot_ids)
```

Reply:
e.g. cant you implement one function as a function of the other?

Reply:
ääh yes.. I think so. not sure what I thought here.